docs/develop/typescript/data-encryption.mdx

```diff
@@ -11,41 +11,129 @@ tags:
 
-A Data Converter has two parts:
+## Payload Codec
 
-- [Payload Converter](#payload-converter): Sync methods that sometimes run inside the Workflow isolate (and are thus limited).
-- [Payload Codec](#payload-codec): Async methods that run outside the isolate.
+> API documentation: [PayloadCodec](https://typescript.temporal.io/api/interfaces/common.PayloadCodec)
+
+The default `PayloadCodec` does nothing. To create a custom one, we implement the following interface:
 
 ```ts
-interface DataConverter {
-  payloadConverterPath?: string;
-  payloadCodecs?: PayloadCodec[];
+interface PayloadCodec {
+  /**
+   * Encode an array of {@link Payload}s for sending over the wire.
+   * @param payloads May have length 0.
+   */
+  encode(payloads: Payload[]): Promise<Payload[]>;
+
+  /**
+   * Decode an array of {@link Payload}s received from the wire.
+   */
+  decode(payloads: Payload[]): Promise<Payload[]>;
 }
 ```
 
-### Payload Converter
+## Use custom payload conversion
 
-> API documentation: [PayloadConverter](https://typescript.temporal.io/api/interfaces/common.PayloadConverter)
+Temporal SDKs provide a [Payload Converter](/dataconversion#payload-converter) that can be customized to convert a custom data type to a [Payload](/dataconversion#payload) and back.
 
-```ts
+The order in which your encoding Payload Converters are applied depending on the order given to the Data Converter.
+You can set multiple encoding Payload Converters to run your conversions.
+When the Data Converter receives a value for conversion, the value gets passes through each Payload Converter in sequence until the converter that handles the data type does the conversion. You will explore more in detail now. You will explore more in detail now.
+
+## Composite Data Converters
+
+Use a [Composite Data Converter](https://typescript.temporal.io/api/classes/common.CompositePayloadConverter) to apply custom, type-specific Payload Converters in a specified order. Defining a new Composite Data Converter is not always necessary to implement custom data handling. You can override the default Converter with a custom Codec, but a Composite Data Converter may be necessary for complex Workflow logic.
+
+A Composite Data Converter can include custom rules created, and it can also
+leverage the default Data Converters built into Temporal. In fact, the default
+Data Converter logic is implemented internally in the Temporal source as a
+Composite Data Converter. It defines these rules in this order:
+
+```typescript
+export class DefaultPayloadConverter extends CompositePayloadConverter {
+  constructor() {
+    super(
+      new UndefinedPayloadConverter(),
+      new BinaryPayloadConverter(),
+      new JsonPayloadConverter()
+    );
+  }
+}
+```
+
+The order of applying the Payload Converters is important. During
```

Comment:
Under what circumstances is the payload null? What does that tell you? How do you handle it?